engine/console/game.go

```diff
@@ -0,0 +1,231 @@
+		col++
+		if col >= x2 {
+			row++
+			col = x1
+		}
+		if row > y2 {
+			break
+		}
+	}
+}
+
+func drawInitialBox(s tcell.Screen, boundary Boundary, style tcell.Style) {
+	x1, y1 := boundary.topLeft.X, boundary.topLeft.Y
+	x2, y2 := boundary.bottomRight.X, boundary.bottomRight.Y
+	if y2 < y1 {
+		y1, y2 = y2, y1
+	}
+	if x2 < x1 {
+		x1, x2 = x2, x1
+	}
+
+	// Fill background
+	for row := y1; row <= y2; row++ {
+		for col := x1; col <= x2; col++ {
+			s.SetContent(col, row, tcell.RuneBullet, nil, style)
+		}
+	}
+
+	// Draw borders
+	for col := x1; col <= x2; col++ {
+		s.SetContent(col, y1, tcell.RuneHLine, nil, style)
+		s.SetContent(col, y2, tcell.RuneHLine, nil, style)
+	}
+	for row := y1 + 1; row < y2; row++ {
+		s.SetContent(x1, row, tcell.RuneVLine, nil, style)
+		s.SetContent(x2, row, tcell.RuneVLine, nil, style)
+	}
+
+	// Only draw corners if necessary
+	if y1 != y2 && x1 != x2 {
+		s.SetContent(x1, y1, tcell.RuneULCorner, nil, style)
+		s.SetContent(x2, y1, tcell.RuneURCorner, nil, style)
+		s.SetContent(x1, y2, tcell.RuneLLCorner, nil, style)
+		s.SetContent(x2, y2, tcell.RuneLRCorner, nil, style)
+	}
+
+	//drawText(s, x1+1, y1+1, x2-1, y2-1, style, text)
+}
+
+func drawSnake(s tcell.Screen, snake Snake, boundary Boundary) error {
+	if boundary.Contains(snake.head) {
+		return fmt.Errorf("snakep2p's head coordinates are out of boundary")
+	}
+	s.SetContent(snake.head.X, snake.head.Y, tcell.RuneBullet, nil, snake.style)
+	for _, point := range snake.body {
+		if boundary.Contains(point) {
+			return fmt.Errorf("snakep2p's body coordinates are out of boundary")
+		}
+		s.SetContent(point.X, point.Y, tcell.RuneBlock, nil, snake.style)
+	}
+	return nil
+}
+
+func drawFood(s tcell.Screen, food core.Coord, style tcell.Style, boundary Boundary) error {
+	if boundary.Contains(food) {
+		return fmt.Errorf("food coordinates are out of boundary")
+	}
+	s.SetContent(food.X, food.Y, '*', nil, style)
+	return nil
+}
+
+func drawGridCell(s tcell.Screen, cell core.Coord, style tcell.Style, boundary Boundary) error {
+	if boundary.Contains(cell) {
+		return fmt.Errorf("cell is out of boundary")
+	}
+	s.SetContent(cell.X, cell.Y, tcell.RuneBullet, nil, style)
+	return nil
+}
+
+func getRandColor(defColors map[tcell.Color]struct{}) tcell.Color {
+	color := tcell.PaletteColor(rand.Intn(256))
+	_, ok := defColors[color]
+	for ok == true {
+		color = tcell.PaletteColor(rand.Intn(256))
+		_, ok = defColors[color]
+	}
+	return color
+}
+
+func genSnakeStyle(defColors map[tcell.Color]struct{}) tcell.Style {
+	style := tcell.StyleDefault.Foreground(getRandColor(defColors)).Background(getRandColor(defColors))
+	return style
+}
+
+var defColors = map[tcell.Color]struct{}{
+	tcell.ColorReset: {},
+	tcell.ColorWhite: {},
+	tcell.ColorPurple: {},
+	tcell.ColorRed: {},
+}
+
+func (game *Game) handleGameEvent(event interface{}) {
+	switch event := event.(type) {
+	case core.PlayerStarts:
+		(*game).numAliveSnakes = len(event.Players)
```

Comment:
`(*game).numAliveSnakes` and `game.numAliveSnakes` is equivalent. Replace with the later version, please.